packages/compass-components/src/components/bson-value.tsx

```diff
@@ -25,36 +27,27 @@ type PropsByValueType<V extends ValueTypes> = Omit<
   'type'
 >;
 
-export const VALUE_COLOR_BY_TYPE: Record<
-  Extract<
-    TypeCastTypes,
-    | 'Int32'
-    | 'Double'
-    | 'Decimal128'
-    | 'Date'
-    | 'Boolean'
-    | 'String'
-    | 'ObjectId'
-  >,
-  string
-> = {
-  // NOTE: these colors are aligned with the Leafygreen code component
-  // and are not part of the palette
-  Int32: '#016ee9',
-  Double: '#016ee9',
-  Decimal128: '#016ee9',
-  Date: '#016ee9',
-  Boolean: '#CC3887',
-  String: '#12824D',
-  ObjectId: '#D83713',
+const VALUE_COLOR_BY_THEME_AND_TYPE: Record<Theme, Record<string, string>> = {
```

Comment:
Any reason for making the types less accurate here? This could at least be
```suggestion
const VALUE_COLOR_BY_THEME_AND_TYPE: Record<Theme, Partial<Record<TypeCastTypes, string>>> = {
```